src/lib.rs

```diff
@@ -9,11 +9,45 @@
 //!
 //! * [MIT license](http://opensource.org/licenses/MIT)
 use proc_macro::TokenStream;
-use proc_macro2::TokenStream as TokenStream2;
-use quote::{format_ident, quote};
+use proc_macro2::{Span, TokenStream as TokenStream2};
+use quote::{format_ident, quote, quote_spanned, ToTokens, TokenStreamExt};
 use std::collections::hash_map::DefaultHasher;
 use std::hash::{Hash, Hasher};
-use syn::{parse_macro_input, DeriveInput, Type};
+use syn::spanned::Spanned;
+use syn::{parse_macro_input, parse_quote, Field, Ident, LitStr, Path, Type};
+
+use darling::FromDeriveInput;
+
+fn default_crate_root() -> Path {
+    parse_quote!(::bpf_script)
+}
+
+#[derive(FromDeriveInput)]
+#[darling(supports(struct_named))]
```

Comment:
The macro tells the compiler it wants `#[field(...)]` attributes, but then doesn't use them. I omitted that from the `darling` declaration for now.